### PR TITLE
linkAndJunctionsParsingFromOSM

### PR DESCRIPTION
**Pull request by An7ar35**
Huge update that include stuff relating to Graph, OSM, GUI, beahviour...

### CODE REVIEW

**Comment by An7ar35** [comment]
@lexaux Can't assign more than one person.. I'll check your bits and you check mine I guess.

**Comment by lexaux** [comment]
That's the plan.

**Comment by An7ar35** [comment]
OK, so that's weird - the tick test on TrafficGenerator fails here but passes through gradle locally.

File: gui/src/main/java/kcl/teamIndexZero/traffic/gui/components/ChooserDialog.java [hunk 1]
```diff
@@ -85,4 +139,36 @@ public static void showForOSMLoadResult(Consumer<OsmParseResult> resultConsumer)
         new ChooserDialog(resultConsumer).setVisible(true);
     }
 
+    public void openFile(String file, String caption) {
+        try {
+            InputStream stream;
+            if (file.endsWith(".gz")) {
```
Comment (lexaux) [inline gui/src/main/java/kcl/teamIndexZero/traffic/gui/components/ChooserDialog.java]: Shall we close stream here?

File: gui/src/main/java/kcl/teamIndexZero/traffic/gui/components/SimulationWindow.java [hunk 2]
```diff
@@ -41,9 +41,15 @@ public SimulationWindow(GuiModel model, GuiController controller) {
         setExtendedState(getExtendedState() | JFrame.MAXIMIZED_BOTH);
         setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
         getContentPane().setLayout(new BorderLayout());
+
+        JPanel detailsPanel = new SimulationDetailsPanel(model, controller);
+        detailsPanel.setMaximumSize(new java.awt.Dimension(400, 32767));
```
Comment (lexaux) [inline gui/src/main/java/kcl/teamIndexZero/traffic/gui/components/SimulationWindow.java]: strange number

File: simulator/src/main/java/kcl/teamIndexZero/traffic/simulator/data/GraphConstructor.java [hunk 3]
```diff
@@ -80,95 +76,13 @@ public GraphConstructor(List<JunctionDescription> junction_descriptions, List<Ro
...
-                //check if traffic generator has been visited.
-                //> if not switch to visited and return true;
-            } else {
-                //NOT Supported!
-                //LOG fatal
-                //throw exception
-            }
-        }
-        return false;
+    private void createGraph(List<JunctionDescription> junction_descriptions, List<RoadDescription> road_descriptions) throws MapIntegrityException {
```
Comment (lexaux) [inline simulator/src/main/java/kcl/teamIndexZero/traffic/simulator/data/GraphConstructor.java]: That's quite cleaned up, thanks.

File: simulator/src/main/java/kcl/teamIndexZero/traffic/simulator/data/features/Lane.java [hunk 4]
```diff
@@ -4,29 +4,91 @@
...
+        return indexInDirectedLanes;
+    }
+
+    public GeoPolyline getPolyline() {
+        return polyline;
+    }
+
+    /**
+     * Externally called by road constructor to create a polyline. We need to have all lanes created at this moment.
+     */
```
Comment (lexaux) [inline simulator/src/main/java/kcl/teamIndexZero/traffic/simulator/data/features/Lane.java]: Need better explanation.

**Comment by lexaux** [comment]
@An7ar35 I looked it it and said: behold you the people, this is what the good is. merging.

**Comment by An7ar35** [comment]
Nice :)

**Pull request merged**